cmd/common.go

```diff
@@ -234,12 +277,27 @@ func syncMain(ctx context.Context, filenames []string, dry bool, parallelism,
 	totalOps, err := performDiff(
 		ctx, currentState, targetState, dry, parallelism, delay, kongClient, mode == modeKonnect)
 	if err != nil {
-		return err
+		if isJSONOutput {
+			if errs, ok := err.(utils.ErrArray); ok {
+				jsonOutput.Errors = append(jsonOutput.Errors, errs.ErrorList()...)
+			} else {
+				jsonOutput.Errors = append(jsonOutput.Errors, fmt.Sprintf("%v", err))
+			}
+		} else {
+			return err
+		}
 	}
-
 	if diffCmdNonZeroExitCode && totalOps > 0 {
 		os.Exit(exitCodeDiffDetection)
 	}
+	if enableJSONOutput {
+		jsonOutputStr, jsonErr := json.MarshalIndent(jsonOutput, "", " ")
```

Comment:
This is not a string yet.

```suggestion
		jsonOutputBytes, jsonErr := json.MarshalIndent(jsonOutput, "", " ")
```